packages/cli/src/util/paths.ts

```diff
@@ -0,0 +1,179 @@
+import { generateHttpParam } from '@stoplight/prism-http';
+import { serializeWithDeepObjectStyle } from '@stoplight/prism-http';
+import { serializeWithPipeDelimitedStyle, serializeWithSpaceDelimitedStyle } from '@stoplight/prism-http';
+import {
+  Dictionary,
+  HttpParamStyles,
+  IHttpOperation,
+  IHttpParam,
+  IHttpPathParam,
+  IHttpQueryParam,
+  INodeExample,
+  INodeExternalExample,
+} from '@stoplight/types';
+import * as Either from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get } from 'lodash';
+// @ts-ignore
+import { parse } from 'uri-template';
+
+export function createExamplePath(operation: IHttpOperation): Either.Either<Error, string> {
+  return pipe(
+    generateTemplateAndValuesForPathParams(operation),
+    Either.chain(({ template: pathTemplate, values: pathValues }) => {
+      return pipe(
+        generateTemplateAndValuesForQueryParams(pathTemplate, operation),
+        Either.map(({ template: queryTemplate, values: queryValues }) => {
+          return { template: queryTemplate, values: { ...pathValues, ...queryValues } };
+        }),
+      );
+    }),
+    Either.map(({ template, values }) => parse(template).expand(values)),
+  );
+}
+
+function generateParamValue(spec: IHttpParam): Either.Either<Error, unknown> {
+  return pipe(
+    generateHttpParam(spec),
+    Either.fromOption(() => new Error(`Cannot generate value for: ${spec.name}`)),
+    Either.chain(value => {
+      switch (spec.style) {
+        case HttpParamStyles.DeepObject:
+          return Either.right(serializeWithDeepObjectStyle(spec.name, value));
+
+        case HttpParamStyles.PipeDelimited:
+          if (Array.isArray(value)) {
+            return Either.right(
+              serializeWithPipeDelimitedStyle(spec.name, value as Array<string | number | boolean>, spec.explode),
+            );
+          } else {
+            return Either.left(new Error('Pipe delimited style is only applicable to array parameter'));
+          }
+
+        case HttpParamStyles.SpaceDelimited:
+          if (Array.isArray(value)) {
+            return Either.right(
+              serializeWithSpaceDelimitedStyle(spec.name, value as Array<string | number | boolean>, spec.explode),
+            );
+          } else {
+            return Either.left(new Error('Space delimited style is only applicable to array parameter'));
+          }
```

Comment:
ditto, you can change this with

```ts
return pipe(
  value,
  Either.fromPredicate(
    Array.isArray,
    () => new Error('Pipe delimited style is only applicable to array parameter'),
  ),
  Either.map(v => serializeWithPipeDelimitedStyle(spec.name, v, spec.explode)),
);
```

You can also see that this code is very similar to the one handling the other case; if we could DRY it and reuse that's be even better.

Reply:
There are two differences between them. Error message and called function. I'd say it would be overengineering to make a common function for that and definitely would decrease readability.

Reply:
I'm ok keeping it in this way.